File: Python/424A-squats.py
n = int(input())
pos = input()
x = []
X = []

for i in range(n):
    x.append(i) if pos[i] == 'x' else X.append(i)

c = len(x) - len(X)
if c > 0:
    for i in range(c // 2):
        pos = pos[:x[i]] + 'X' + pos[x[i] + 1:]
elif c < 0:
    for i in range(-c // 2):
        pos = pos[:X[i]] + 'x' + pos[X[i] + 1:]

print(abs(c // 2))
print(pos)

File: C++/A/424A-squats.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    string pos;
    cin >> n >> pos;
    vector<int> x, X;
    for (int i = 0; i < n; i++)
        (pos[i] == 'x') ? x.push_back(i) : X.push_back(i);
    int c = x.size() - X.size();
    if (c > 0)
        for (int i = 0; i < c / 2; i++)
            pos[x[i]] = 'X';
    else if (c < 0)
        for (int i = 0; i < abs(c / 2); i++)
            pos[X[i]] = 'x';

    cout << abs(c / 2) << '\n'
         << pos;
}